### DISCUSSION TITLE
Client Session Idle Not Triggering Expiration of Session

### DISCUSSION THREAD

**Discussion by reedpatterson** [keycloak/keycloak#32867] [Q&A]
Can somebody help me understand Client Session Idle? I am using the angular oauth oidc2 library; to my understanding, Client Session Idle is an inactivity timeout that -- when that oauth library does not interface with keycloak for a certain number of minutes (1 minute, in my case for testing), the session should expire.

![image](https://github.com/user-attachments/assets/eddc5eae-fca4-4e36-8734-5c9ab90690e8)

In my case, I sign into my webapp and immediately close the page. Since the application is not open, the browser should not be firing oauth events, and after a minute (technically three minutes due to keycloaks buffer? https://keycloak.discourse.group/t/session-duration-when-the-user-is-not-active/17811) I would expect the session to expire.

This does not happen to be the case, however. After several minutes later (10+), I am able to open a new tab, navigate to my webapp, and log in without being redirecting to KC SSO page. Furthermore, looking at my list of sessions on keycloak, I see that the session is not terminated. For some reason, the client disappeared under the specific session under "Clients" (should be 'ng-dev-reed'), despite still being active.

![image](https://github.com/user-attachments/assets/c43f4f6e-86d5-463a-a952-0098ddc129d6)
![image](https://github.com/user-attachments/assets/74890825-5ec3-4091-9333-ebbe7b54c97c)

If anyone can explain this behavior to me, that would be greatly appreciated, thank you.

**Comment by takehana-ke** [comment]
@reedpatterson 
I also faced the same issue.

It seems that this configuration is specific to the service account and is not related to the user's session.  
In fact, [the document](https://www.keycloak.org/docs/latest/server_admin/index.html#_timeouts) also mentions the following:

> This setting never affects the general SSO user session, which is unique.
